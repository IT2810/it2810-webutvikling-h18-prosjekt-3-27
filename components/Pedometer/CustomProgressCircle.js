import React, {Component} from "react";
import {View} from "react-native";
import ProgressCircle from 'react-native-progress/Circle';

export default class CustomProgressCircle extends Component {

  getColor = () => {
    const colors = ["#E50100", "#E02F00","#DC5B00","#D88600","#D4AE00","#CACF00","#9CCB00","#70C800","#46C300","#1DBF00"];
    let p = this.props.progress;
    console.log(p);
    if (p <= 0.1) {
      return colors[0];
    } else if (p > 0.1 && p <= 0.2) {
      return colors[1];
    } else if (p > 0.2 && p <= 0.3) {
      return colors[2];
    } else if (p > 0.3 && p <= 0.4) {
      return colors[3];
    } else if (p > 0.4 && p <= 0.5) {
      return colors[4];
    } else if (p > 0.5 && p <= 0.6) {
      return colors[5];
    } else if (p > 0.6 && p <= 0.7) {
      return colors[6];
    } else if (p > 0.7 && p <= 0.8) {
      return colors[7];
    } else if (p > 0.8 && p <= 0.9) {
      return colors[8];
    } else if (p > 0.9) {
      return colors[9];
    }
  };

  render() {
    return(
      <View>
        <ProgressCircle
          progress={isNaN(this.props.progress) ? 0 : this.props.progress}
          size={250}
          thickness={10}
          color={this.getColor()}
          showsText={true}/>
      </View>
    )
  }
}
