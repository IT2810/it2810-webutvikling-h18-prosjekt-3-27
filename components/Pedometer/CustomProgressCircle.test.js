import React from 'react';
import renderer from 'react-test-renderer';

import CustomProgressCircle from "./CustomProgressCircle";


test('renders correctly', () => {
  const tree = renderer.create(<CustomProgressCircle progress={1}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
