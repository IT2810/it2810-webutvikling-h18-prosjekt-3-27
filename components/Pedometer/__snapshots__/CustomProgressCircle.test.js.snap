// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <ARTSurfaceView
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          },
          Object {
            "height": 250,
            "width": 250,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            125,
            6,
            4,
            125,
            125,
            119,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            125,
            125,
            119,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke={
          Array [
            0.11372549019607843,
            0.7490196078431373,
            0,
            1,
          ]
        }
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={10}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            125,
            0.5,
            4,
            125,
            125,
            124.5,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            125,
            125,
            124.5,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke={
          Array [
            0.11372549019607843,
            0.7490196078431373,
            0,
            1,
          ]
        }
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTSurfaceView>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 114,
          "height": 228,
          "justifyContent": "center",
          "left": 11,
          "position": "absolute",
          "top": 11,
          "width": 228,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#1DBF00",
              "fontSize": 50.666666666666664,
              "fontWeight": "300",
            },
            undefined,
          ]
        }
      >
        0%
      </Text>
    </View>
  </View>
</View>
`;
